src/common/model/kind/kindVersion.ml

```diff
@@ -0,0 +1,127 @@
+open Nes
+
+let _key = "kind-version"
+
+type t = int * KindBase.t
+
+let to_string (repeats, base) =
+  spf "%d %s" repeats (KindBase.to_string base)
+
+let of_string s =
+  let s = NesString.remove_char ' ' s in
+  try
+    ssf s "%d%[a-zA-Z]"
+      (fun repeats base -> (repeats, KindBase.of_string base))
+  with
+  | End_of_file | Scanf.Scan_failure _ ->
+    try
+      ssf s "%[a-zA-Z]%d"
+        (fun base repeats -> (repeats, KindBase.of_string base))
+    with
+    | End_of_file | Scanf.Scan_failure _ ->
+      invalid_arg "Dancelor_common_model.Kind.version_of_string"
+
+let%test _ = to_string (32, Waltz) = "32 W"
+let%test _ = to_string (64, Reel) = "64 R"
+let%test _ = to_string (24, Jig) = "24 J"
+let%test _ = to_string (48, Strathspey) = "48 S"
+
+let%test _ = of_string "W 32" = (32, Waltz)
+let%test _ = of_string "64 Reel" = (64, Reel)
+let%test _ = of_string "JIG 24" = (24, Jig)
+let%test _ = of_string "48 sTrathPEY" = (48, Strathspey)
```

Comment:
How does this test work despite the missing "s" ?

Reply:
Hah, that's a funny remark and a funny test case. I think we only use the first letter (cf `KindBase.ml` line 31), but that's maybe a bit weak and we'd prefer enforcing either one letter or the whole correct word?

Reply:
Logged in https://github.com/paris-branch/dancelor/issues/227.